doc/nat/nat_design_spec.md

```diff
@@ -0,0 +1,1550 @@
+
+The allowed range of configurable zone values is 0 to 3.
+
+In typical DC deployments, the Loopback interface IP address can be used as the public IP address in the static or dynamic NAT configurations. In such a case, Loopback interface as well should be assigned a zone value that is same as the zone assigned to the physical outside interfaces. Though the zone value is not configured for the Loopback interface in the hardware, it is required by the application layer for the NAT functionality.
+
+More details on the configuration of the Loopback IP as public IP is in the section 3.4 below.
+
+### 2.2.5 Twice NAT and NAPT
+Twice NAT or Double NAT is a NAT variation where both the Source IP and the Destination IP addresses are modified as a packet crosses the address zones. It is typically used in  the communication between networks with overlapping private addresses.
+
+The configuration for Twice NAT/NAPT is achieved in 2 ways:
+- Putting two Static NAT/NAPT entries in the same Group ('twice_nat_id' value).
+- Putting a Dynamic NAT/NAPT Binding and a Static type NAT/NAPT entry in the same Group ('twice_nat_id' value).
+
+When a host matching a dynamic NAT pool binding sends traffic to host with a matching DNAT Static NAT/NAPT entry in the same 'twice_nat_id' group, a bi-directional Twice NAT/NAPT entry is created for the traffic flow.
+
+NOTE: The Static NAT/NAPT entry that is part of a Twice NAT group is not added used for single NAT'ting in the hardware.
+
+### 2.2.6 VRF support
+NAT is supported in the default Virtual Routing and Forwarding (VRF) domain only. 
+It is assumed that the Static or Dynamic NAT configurations do not include the interfaces in non-default VRFs.
+
+# 3 Design
+## 3.1 Design overview
+The design overview at a high level is as below. The details are explained in the following subsections.
+- Configuration tables and schema for provisioning the static and dynamic NAT/NAPT entries.
+- Design is centered around the iptables and connection tracking modules in the kernel. 
+- NAT applications configures the NAT configuration rules using iptables in the kernel.
+- The iptables and connection tracking modules in the kernel are leveraged to handle the NAT miss packets to create the NAT/NAPT entries, that are added to the hardware.
+
+## 3.2 DB Changes
+### 3.2.1 Config DB
+Following table changes are done in Config DB.
+
+#### 3.2.1.1 STATIC_NAPT Table
+```
+STATIC_NAPT|{{global_ip}}|{{ip_protocol}}|{{global_l4_port}} 
+    "local_ip": {{ip_address}}
+    "local_port" :{{l4_port}}
+    "nat_type"   :{{snat-or-dnat}}      (OPTIONAL)
+    "twice_nat_id" :{{twice-nat-id}}    (OPTIONAL)
+```
+
+```
+This config tells to do:
+- If the "nat_type" is 'dnat':
+  - DNAT translation of the DIP/DPORT in the IP packet from 'global_ip' address and 'global_port' to 'local_ip' address and 'local_l4_port' when the packet crosses the zones.

```

Comment:
This is meant to say, DNAT happens from the configured values of global to local ip addresses, when the matching DNAT entry exists in the hardware tables. Crossing the zones statement is no longer required here. Shall remove the same. Since zones can still be equal in case of hair-pin nat'ting case.

Reply:
@kirankella, the corresponding section in doc seems not yet updated.

Reply:
This section is already updated for the DNAT in the latest HLD. Crossing the zones statement is valid for SNAT. It is not valid for DNAT.